core/server/types/helmrepository_test.go

```diff
@@ -34,7 +34,7 @@ func TestHelmRepository(t *testing.T) {
 			&pb.HelmRelease{
 				HelmChart: &pb.HelmChart{
 					Name: "-",
-					SourceRef: &pb.SourceRef{
+					SourceRef: &pb.ObjectRef{
 						Kind: -1, // This is invalid?
```

Comment:
Not related to your changes, but this can be `pb.FluxObjectKind_KindHelmRepository` instead of -1:
```
				HelmChart: &pb.HelmChart{
					Name: "-",
					SourceRef: &pb.ObjectRef{
						Kind: pb.FluxObjectKind_KindHelmRepository,
					},
				},
```

Reply:
Yeah, it's a bit weird - https://github.com/weaveworks/weave-gitops/blob/main/core/server/types/utils.go#L11 seems to generate schema-violating values if the input is "unexpected". I wonder if it'd be worth adding a FluxObjectKind value which is Unknown, so nothing crashes in these kind of cases :thinking: